osu.Game/Overlays/News/Sidebar/YearsPanel.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using System.Collections.Generic;
+using osu.Game.Graphics;
+using osu.Framework.Bindables;
+using osu.Game.Online.API.Requests.Responses;
+using System;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays.News.Sidebar
+{
+    public class YearsPanel : CompositeDrawable
+    {
+        private readonly Bindable<APINewsSidebar> metadata = new Bindable<APINewsSidebar>();
+
+        private Container gridPlaceholder;
+
+        [BackgroundDependencyLoader]
+        private void load(OverlayColourProvider colourProvider, Bindable<APINewsSidebar> metadata)
+        {
+            this.metadata.BindTo(metadata);
+
+            AutoSizeAxes = Axes.Y;
+            RelativeSizeAxes = Axes.X;
+            Masking = true;
+            CornerRadius = 6;
+            InternalChildren = new Drawable[]
+            {
+                new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = colourProvider.Background3
+                },
+                gridPlaceholder = new Container
+                {
+                    RelativeSizeAxes = Axes.X,
+                    AutoSizeAxes = Axes.Y,
+                    Padding = new MarginPadding(5)
+                }
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            metadata.BindValueChanged(m =>
+            {
+                if (m.NewValue == null)
+                {
+                    Hide();
+                    return;
+                }
+
+                gridPlaceholder.Child = new YearsGridContainer(m.NewValue.Years, m.NewValue.CurrentYear);
+                Show();
+            }, true);
+        }
+
+        private class YearButton : OsuHoverContainer
+        {
+            protected override IEnumerable<Drawable> EffectTargets => new[] { text };
```

Comment:
Unnecessary?

Reply:
There's a slight colour change on hover

Reply:
Yes, but check the base implementation. I'm saying this line is unnecessary to keep that effect working.

Reply:
Oh, misunderstood the comment